jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -473,6 +475,11 @@ object XGBoost extends Serializable {
             tracker.stop()
           }
       }.last
+      // we should delete the checkpoint directory after a successful training
+      if (!skipCleanCheckpoint) {
```

Comment:
[Really minor] I am not sure whether xgboost has java|scala coding style. I generally prefer to have `cleanCheckpoint` flag not `skipCleanCheckpoint`. That would avoid adding extra "not" which makes reading the code slightly harder
P.S. This is really bike shadding

Reply:
as explained above, cleaning checkpoint is the desired behavior. skipCleanCheckpoint is there mainly for testing

regarding the use cases you want to continue training, I think that belongs to another feature that you can start training from an existing model

Reply:
Agree on your proposed feature: train starting from an existing model